engine/engine.py

```diff
@@ -87,7 +87,27 @@ class NewEngineNodeTree(bpy.types.Operator):
     bl_label = "New Node Tree"
 
     def execute(self, context):
-        bpy.ops.node.new_node_tree(type="DreamTexturesNodeTree")
+        # TODO: Get the name of the default node tree from a config file
+        # TODO: type is deprecated https://docs.blender.org/api/current/bpy.types.NodeTree.html 
+        #       When testing the type of the resulting node tree using  bpy.data.node_groups['Dream Textures Node Tree'].type it is '' because of that
+        bpy.ops.node.new_node_tree(type="DreamTexturesNodeTree", name="Dream Textures Node Editor")
+        # Get the newly generated node tree
+        node_tree = bpy.data.node_groups[-1]
+        node_sd = node_tree.nodes.new(type="dream_textures.node_stable_diffusion")
+        node_sd.location = (200, 200)
+        node_out = node_tree.nodes.new(type="NodeGroupOutput")
+        node_tree.outputs.new('NodeSocketColor','Image')
```

Comment:
Missed this before, `outputs` and `inputs` were removed in Blender 4. Could you add a check here for the `bpy.app.version`, and in Blender 4.0 use the [`.interface.new_socket(...)`](https://docs.blender.org/api/current/bpy.types.NodeTreeInterface.html) API?

Reply:
Added this to latest iteration but was not able to test yet.